source/tutorials/cross-compilation.rst

```diff
@@ -0,0 +1,257 @@
+.. _ref-cross-compilation:
+
+Cross-compilation
+=================
+
+Cross-compilation is the act of **compiling code** on the **build platform**
+to the **host platform**, where the compiled **executable runs**. [#]_
+
+It's needed when the host platform has limited resources (such as CPU)
+or when it's not easily accessible for development.
+
+Nix community has world-class support for cross-compilation,
+after years of hard work from our community.
+
+.. [#] Terminology for cross-compilation platforms differs between build systems,
+       Nix community has chosen to follow 
+       `autoconf terminology <https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Hosts-and-Cross_002dCompilation.html>`_.
+
+.. note:: macOS/Darwin is a special case, as not the whole OS is Open Source. 
+          It's only possible to cross-compile between ``aarch64-darwin`` and ``x86_64-darwin``.
+
+
+What's a target platform?
+-------------------------
+
+There's actually a third platform named target.
+
+It matters in cases where you'd like to distribute a compiler binary, 
+as you'd then like to build a compiler on the build platform, compile code on the
+target plaform and run the final executable on the host platform.
+
+Since that's rarely needed, we'll treat target platform the same as the build.
```

Comment:
Just write that you ignore the platform for this tutorial. The default case of targetPlatform is the same as hostPlatform afaik.

Also interesting to note: On the outermost interface nixpkgs doesn't support `targetPlatform`, it only accepts `localSystem` (~ build) and `crossSystem` (~ host) since cross compiling a cross compiler is a rare use case outside of stdenv bootstrapping.

Reply:
I'd like to clarify since it took me quite some time to figure out why target is not important and explain why it can be ignored to save confusion. The user will eventually stumble upon it as they dive into cross-compilation.

Reply:
I've thought about this yesterday and maybe it would actually *help* the tutorial to explain target completely. This would make it very easy to clear the following things up:

* Why are binutils, pkg-config and cc prefixed? What does the prefix mean? This is quite easy to answer if you explain the relation of the target prefix to the target platform
* What is `buildPackages`? Here having an understanding of target is nice: `buildPackages` usually has build and host platform equivalent to `localSystem`, but its target is `crossSystem` — having an understanding of target automatically explains how we bootstrap our cross package sets

Reply:
Agreed! I don't think I know enough to really explain those, unfortunately.